keycloak/user.go

```diff
@@ -44,3 +47,69 @@ func (keycloakClient *KeycloakClient) UpdateUser(user *User) error {
 func (keycloakClient *KeycloakClient) DeleteUser(realmId, id string) error {
 	return keycloakClient.delete(fmt.Sprintf("/realms/%s/users/%s", realmId, id))
 }
+
+func (keycloakClient *KeycloakClient) getUserByUsername(realmId, username string) (*User, error) {
+	var users []*User
+
+	err := keycloakClient.get(fmt.Sprintf("/realms/%s/users?username=%s", realmId, url.QueryEscape(username)), &users)
+	if err != nil {
+		return nil, err
+	}
+
+	// more than one user could be returned so we need to search through all of the results and return the correct one
+	// ex: foo and foo-user could both exist, but searching for "foo" will return both
+	for _, user := range users {
+		if user.Username == username {
+			user.RealmId = realmId
+
+			return user, nil
+		}
+	}
+
+	// the requested user does not exist
+	// I don't think raising an error is appropriate here - consumers should check if the user is nil
```

Comment:
> //  I don't think raising an error is appropriate here - consumers should check if the user is nil

 This comment is a little strange because the method is private to the package, so all the consumers are in the package.